### PR TITLE
removed deprecated inf and nan

### PR DESCRIPTION
**Pull request by zenna**
I was getting many warnings when using `quantile` due to

Inf(Float32/64)
nan(Float32/64)

### CODE REVIEW

File: src/specialfuns.jl [hunk 1]
```diff
@@ -116,16 +116,16 @@ for (fn,arg) in ((:Φinv,:p),(:logΦinv,:logp))
             else
                 if $(fn == :Φinv)
                     if p <= 0f0
-                        return p == 0f0 ? -inf(Float32) : nan(Float32)
+                        return p == 0f0 ? -Inf32 : NaN32
                     elseif p >= 1f0 
-                        return p == 1f0 ? inf(Float32) : nan(Float32)
+                        return p == 1f0 ? inf(Float32) : NaN32
```
Comment (simonster) [inline src/specialfuns.jl]: `inf(Float32) -> Inf32` here as well?

File: src/specialfuns.jl [hunk 2]
```diff
@@ -116,16 +116,16 @@ for (fn,arg) in ((:Φinv,:p),(:logΦinv,:logp))
...
                     end
                     r = sqrt(q < 0f0 ? -log(p) : -log1p(-p))
                 else
                     if logp == -Inf
-                        return -inf(Float32)
+                        return -Inf32
                     elseif logp >= 0f0 
-                        return logp == 0f0 ? inf(Float32) : nan(Float32)
+                        return logp == 0f0 ? inf(Float32) : NaN32
```
Comment (simonster) [inline src/specialfuns.jl]: and here

**Comment by zenna** [comment]
This isn't failing because of me, I don't think.  Some library is using Range1 which it seems was recently removed from Julia.

**Comment by simonster** [comment]
Merged. But we should really fix that Range1 issue...

**Comment by simonster** [comment]
This is actually passing for me locally, so I suspect that one of the dependencies needs to be bumped in METADATA.

**Comment by zenna** [comment]
It might be ArrayViews

**Comment by simonster** [comment]
I think you're right, or at least ArrayViews definitely needed to be bumped, and I just did that. We'll see if it builds now.

**Comment by simonster** [comment]
Also StatsBase.

**Pull request merged**